internal/testutils/wrappers.go

```diff
@@ -72,3 +72,13 @@ func NewListenerWrapper(t *testing.T, lis net.Listener) *ListenerWrapper {
 		NewConnCh: NewChannel(),
 	}
 }
+
+// ErrCloseWrapper wraps closer with a function that does not return an error,
+// but calls t.Error if it does, instead.
+func ErrCloseWrapper(t *testing.T, closer func() error) func() {
```

Comment:
Same here. If the `closer` doesn't return an informative error, we don't have enough contextual information here to push an informative error message to the test output.